Sample Apps/Storefront/Storefront/CartButton.swift

```diff
@@ -26,38 +26,251 @@
 
 import UIKit
 
-class CartButton: RoundedButton {
+class CartButton: UIBarButtonItem {
     
-    override func setTitle(_ title: String?, for state: UIControlState) {
+    private var cartView: CartButtonView!
+
+    // ----------------------------------
+    //  MARK: - Init -
+    //
+    override init() {
+        super.init()
         
-        guard let title = title else {
-            super.setTitle(nil, for: state)
-            return
+        self.initialize()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        
+        self.initialize()
+    }
+    
+    override var customView: UIView? {
+        didSet {
+            if self.customView !== self.cartView {
+                self.customView = self.cartView
+            }
         }
+    }
+    
+    private func initialize() {
+        self.cartView   = self.createCartButtonView()
+        self.customView = self.cartView
         
-        let currentFont  = self.titleLabel!.font!
-        let currentColor = self.titleColor(for: state) ?? .white
+        self.updateBadgeCount(animated: false)
+        self.registerNotifications()
+    }
+    
+    deinit {
+        self.unregisterNotifications()
+    }
+    
+    // ----------------------------------
+    //  MARK: - Notifications -
+    //
+    private func registerNotifications() {
+        NotificationCenter.default.addObserver(self, selector: #selector(itemCountDidChange(_:)), name: Notification.Name.CartControllerItemsDidChange, object: nil)
+    }
+    
+    private func unregisterNotifications() {
+        NotificationCenter.default.removeObserver(self)
+    }
+    
+    private dynamic func itemCountDidChange(_ notification: Notification) {
+        self.updateBadgeCount(animated: true)
+        print("Badge count change notification received")
+    }
+    
+    // ----------------------------------
+    //  MARK: - Cart View -
+    //
+    private func createCartButtonView() -> CartButtonView {
         
-        let style           = NSMutableParagraphStyle()
-        style.alignment     = .center
-        style.lineBreakMode = .byClipping
+        let cartView   = CartButtonView(type: .system)
+        let cartImage  = #imageLiteral(resourceName: "cart")
```

Comment:
Funny enough, you never actually have to type out `#imageLiteral` to get auto-complete. Just typing the image name will achieve the same thing.